CollAction/Frontend/app/global/NewsletterSignup.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from "react";
+import renderComponentIf from "./renderComponentIf";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
+import * as classnames from "classnames";
+
+library.add(faEnvelope);
+
+enum States { VOID, ERROR, READY, LOADING }
+
+interface NewsLetterSignupProps {
+  mailchimpUser: string;
+  mailChimpListId: string;
+}
+
+interface NewsLetterSignupState {
+  subscriptionState: States;
+  email?: string;
+  signupStatusMessage?: string;
+}
+
+class NewsletterSignup extends React.Component<NewsLetterSignupProps, NewsLetterSignupState> {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      subscriptionState: States.VOID
+    };
+
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.getFormContainerClassNames = this.getFormContainerClassNames.bind(this);
+  }
+
+  componentDidMount() {}
+
+  getFormContainerClassNames() {
+    return classnames(
+      "newsletter-signup__container",
+      {
+        "loading": this.state.subscriptionState === States.LOADING,
+        "hidden": this.state.subscriptionState === States.READY,
+      }
+    );
+  }
+
+  getStatusMessageClassNames() {
+    return classnames(
+      "newsletter-signup__message",
+      {
+        "show-error" : this.state.subscriptionState === States.ERROR,
+        "show-success" : this.state.subscriptionState === States.READY
+      }
+    );
+  }
+
+  handleEmailChange(event) {
+    this.setState({
+      email: event.target.value,
+      subscriptionState: States.VOID
+    });
+  }
+
+  isValidEmail() {
+    return /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}/.test(this.state.email);
```

Comment:
Validating the e-mail through a regex isn't a great idea, this will have a lot of false negatives. Maybe only if it contains the '@' symbol? https://davidcel.is/posts/stop-validating-email-addresses-with-regex/

Reply:
Yes, thanks. I will update it.

Reply:
Updated it to a check for just '@' symbol as you suggested @timstokman